docs/resources/stack_dependency_reference.md

```diff
@@ -0,0 +1,51 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "spacelift_stack_dependency_reference Resource - terraform-provider-spacelift"
+subcategory: ""
+description: |-
+  spacelift_stack_dependency_reference represents a Spacelift stack dependency reference - a reference matches a stack's output to another stack's input. It is similar to an environment variable (spacelift_environment_variable), except that value is provided by another stack's output.
+---
+
+# spacelift_stack_dependency_reference (Resource)
+
+`spacelift_stack_dependency_reference` represents a Spacelift **stack dependency reference** - a reference matches a stack's output to another stack's input. It is similar to an environment variable (`spacelift_environment_variable`), except that value is provided by another stack's output.
+
+## Example Usage
+
+```terraform
+resource "spacelift_stack" "infra" {
+  branch     = "master"
+  name       = "Infrastructure stack"
+  repository = "core-infra"
+}
+
+resource "spacelift_stack" "app" {
+  branch     = "master"
+  name       = "Application stack"
+  repository = "app"
+}
+
+resource "spacelift_stack_dependency" "test" {
+  stack_id            = spacelift_stack.app.id
+  depends_on_stack_id = spacelift_stack.infra.id
+}
+
+resource "spacelift_stack_dependency_reference" "test" {
+  stack_dependency_id = spacelift_stack_dependency.test.id
+  output_name         = "DB_CONNECTION_STRING"
+  input_name          = "APP_DB_URL"
```

Comment:
Do we have a way to mark the dependencies as secrets? I'm just thinking for something like this you wouldn't want it being connected as plaintext, and then accidentally leaked.

Reply:
Which variable is sensitive is defined in an output configuration. Reference is only a link output -> input. 
```tf
output "my_secret_output" {
    value = ...
    sensitive = true
}
```

Reply:
Cool - makes sense.